### DISCUSSION TITLE
Run Autogen with Docker

### DISCUSSION THREAD

**Discussion by nhtkid** [microsoft/autogen#469] [Q&A]
Dear community,

I have never used Docker previously. Is there any guide how to run Autogen in Docker in VSCode on Windows?

The example code seems quite simple, which just need to pip install Docker and it will be automatically used. If docker is not present, the code will fall back not to use docker. 

But in reality, it looks quite complicated to me. I need to install Docker desktop, get the Linux image, set up WSL for Windows, etc. By the time I get to VSCode, do I still need to create a virtual environment and then activate a Docker container? Or do I activate a Docker first, then create another virtual environment inside?

Thanks!

**Comment by rickyloynd-microsoft** [comment]
You're right, there are several steps to using Docker on Windows. In general, you don't need to run both Docker and a virtual environment. One or the other is usually enough. Many AutoGen examples will run on Windows directly without WSL/linux. Which example are you trying to run?

**Comment by nhtkid** [comment]
Hi Ricky,

I am running the agents which requires web info.

The example code for Google Colab had docker in it.

So I installed the Docker desktop for Windows, which also automatically installed WSL to my programs and files.

In Vscode, I also pip installed pyautogen and docker inside the virtual environment.

My understanding standing is that the Autogen code will automatically use docker if available without the need to configure the docker. In my case, I just made sure the docker is running from the system tray.

So back to your point that you generally don’t need virtual environment and docker at the same time. What I did is not necessary then? I created a virtual environment in VScode with Python venv command. I also pip installed Docker in that virtual environment.

Thanks for your help!
Nhtkid
________________________________

**Comment by rickyloynd-microsoft** [comment]
Ok, have you tried the basic installation steps and sample code on the main readme page? It should work on windows, WSL, or linux. Which os do you prefer to use?

**Comment by nhtkid** [comment]
Oh yeah.

Rickey I was just copying the exact code example to VSCode on Windows.

Sent from Outlook<http://aka.ms/weboutlook>
________________________________

**Reply by rickyloynd-microsoft** [reply]
Sounds good. Which os are you using? Windows, WSL, linux, or mac?

**Comment by kfsone** [comment]
@nhtkid modern Windows has a built in (but off-by-default) "Sandbox" feature. If you're trying to maintain a pristine environment when trying software out. It's like a gui-capable docker container that's a clone of the OS you're running. if you're not docker familiar, it's a lot easier way to do these things.

win+i -> optional features -> More windowsfeatures -> Windows Sandbox
